src/Engine/MapController.java

```diff
@@ -383,9 +385,12 @@ public void onStateChange()
     switch (inputType)
     {
       case CONSTRAINED_TILE_SELECT:
-        // Create an option selector to keep track of where we are.
-        myGame.setCursorLocation(myGameInputHandler.getCoordinateOptions().get(myGameInputOptionSelector.getSelectionNormalized()));
+      case FREE_TILE_SELECT:
+        if( null != options && options.size() > 0 )
+          myGame.setCursorLocation(options.get(0));
```

Comment:
I'm not a fan of jumping the cursor in free tile select. Also why change to getting `0` instead of `myGameInputOptionSelector.getSelectionNormalized()`?

Reply:
So, this is due to a thing I thought would be fancy: if you're in a target-dense environment (like a Rocket with a block of 3x3 units in range), you get `FREE_TILE_SELECT` instead of constrained, since the way we iterate those is a little unintuitive (and it's always going to be unintuitive for someone)

By the same token, having to move your cursor over to the enemies manually is annoying, so I fixed that too.

As for the selector: it can be null (presumably only the new situations I introduced), and it's always initialized to 0 AFAIK anyway.

Reply:
I get the desire to make target-dense situations a little easier to navigate, but this is no bueno for e.g. Flare and Missile Silo actions? It jumps to the NorthWest-most option, so have fun scrolling back to where you want to be.

Reply:
Bah. Good point.
Maybe this logic should go in `MapController`, then, so it can special-case constrained input as it comes in?

Reply:
This _is_ MapController.

I guess you could use some kind of special-case logic though (only set the cursor for `FREE_TILE_SELECT` if there are fewer than some number of options, maybe?).

Reply:
To clarify:
I was meaning that I would move the "use free select if options are dense" into `MapController`, so that the rocket with friends would say to use constrained, but `MapController` would just interpret input as free select.

Actually actually: Why not (revert this change and) just make Q toggle to free select while a constrained selection is active? Best of both worlds, I think?

Reply:
Maybe, except nobody would know to do that. You'd have to throw up a tooltip somewhere to say how to change modes.

Reply:
Found a better way, I think.
It's almost like `MapController` is the place for control machinations.